accepted/0018-notification-widget.md

```diff
@@ -0,0 +1,35 @@
+- Start Date: 2020-04-17
+- RFC PR: #18
+- Mantis Issue: N/A
+
+- Related PRs:
+	1. https://github.com/obsproject/obs-studio/pull/1832
+
+# Summary
+
+Add the ability to display notifications to the user.
+
+# Motivation
+
+Currenty, there is no way of showing custom notifications to users.
+
+# Detailed design
+
+## User UX
+
+- Display notifications at top of main window
+- Have three notification types (Info, Error, and Warning)
+- Each type will display a different icon and the widget would be a different color for each
+- Display custom text
+- Add button for different types of actions (open source properties, open settings window, open url)
+- The button's text will be customizable
+- The widget will be floating over the window so other widgets aren't resized when it is displayed
```

Comment:
I think this is a double-edged sword, here. It's nice that it doesn't resize the preview, but it's not nice that it obscures the preview. I'm still of a mind that notifications should be placed in the status bar somehow, with an option to click on the notifications to expand them or pop up a larger window.

Reply:
I agree from a informational notification standpoint, but anything important enough to be sent as a warning or error SHOULD disrupt the user and get in the way so that they can't miss it. This isn't something we should be hiding behind an optional click, as I feel that defeats the point.

Reply:
I agree that the notifications should be out, loud, proud, and distracting by design to get user's attention.  Many of OBS's (admittedly limited) UI notifications are, if anything, too subtle for a large chunk of the user base.

That said, however, I do like the idea of having a (non-default) UI option to consolidate the notifications into something like Windows 10's notifications where there's a number and the user can click to get a list of notifications. Bonus points if they're color coded in the list that comes up.

Reply:
Would use the operating systems existing notification systems not be a good place to start with this, both Windows and macOS already have good systems which users are familiar with and would avoid reinventing the wheel over how to design this.